elements/lisk-bft/src/finality_manager.ts

```diff
@@ -368,25 +352,26 @@ export class FinalityManager extends EventEmitter {
 		return true;
 	}
 
-	public reset(): void {
-		this.chainMaxHeightPrevoted = 0;
-	}
-
-	public verifyBlockHeaders(
+	public async verifyBlockHeaders(
 		blockHeader: BlockHeader,
-		bftBlockHeaders: ReadonlyArray<BlockHeader>,
-	): boolean {
+		stateStore: StateStore,
+	): Promise<boolean> {
 		debug('verifyBlockHeaders invoked');
 		debug(blockHeader);
 
+		const bftBlockHeaders = stateStore.consensus.lastBlockHeaders;
+		const { ledger } = await this._getVotingLedger(stateStore);
+		const { preVoted: chainMaxHeightPrevoted } = this._getChainMaxHeightStatus(
+			ledger,
+		);
```

Comment:
Since we already have `getMaxHeightPrevoted`, should not we use it to instead?

Reply:
It is not using because getMaxHeightPrevoted is not using the state store, and during the processing of block, it should use state store consistently